internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -368,48 +369,156 @@ func (m *BlobMetadataManager) getFileOpenCount(filePath string) (int64, error) {
 
 // UpdateHeartbeat creates or updates the heartbeat file
 func (m *BlobMetadataManager) updateHeartbeat(nodeId string, data []byte) error {
-	// Dummy implementation
-	return nil
+	// Create the heartbeat file path
+	heartbeatFilePath := filepath.Join(m.mdRoot, "Nodes", nodeId+".hb")
+	err := m.storageCallback.PutBlobInStorage(internal.WriteFromBufferOptions{
+		Name:                   heartbeatFilePath,
+		Data:                   data,
+		IsNoneMatchEtagEnabled: false,
+		EtagMatchConditions:    "",
+	})
+	if err != nil {
+		log.Err("UpdateHeartbeat :: Failed to put heartbeat blob path %s in storage: %v", heartbeatFilePath, err)
+		// TODO :: Add asserts
```

Comment:
common.Assert(false, fmt.Sprintf("Failed to put heartbeat blob path %s in storage: %v", heartbeatFilePath, err))